staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -223,17 +223,17 @@ func validateAuthInfo(authInfoName string, authInfo clientcmdapi.AuthInfo) []err
 
 		if len(authInfo.ClientCertificate) != 0 {
 			clientCertFile, err := os.Open(authInfo.ClientCertificate)
-			defer clientCertFile.Close()
 			if err != nil {
 				validationErrors = append(validationErrors, fmt.Errorf("unable to read client-cert %v for %v due to %v", authInfo.ClientCertificate, authInfoName, err))
 			}
+			defer clientCertFile.Close()
```

Comment:
this must be under `else`

Reply:
done.

Reply:
By the way, If we need to improve performance here, we can just close file instead of `defer`.